auslib/web/admin/views/rules.py

```diff
@@ -117,10 +118,14 @@ def _post(self, id_or_alias, transaction, changed_by):
 
         what, mapping_values, fallback_mapping_values = process_rule_form(connexion.request.get_json())
 
-        if what.get('mapping', None) is not None and len(mapping_values) != 1:
+        # If 'mapping' key is present in request body but is either blank/null
+        if 'mapping' in what and not what.get('mapping', None):
```

Comment:
I'm still finding these blocks really difficult to read. I think it would help if you expanded them out a bit, and used separate `if` blocks for mapping and fallbackMapping. Eg, something like:
```
if "mapping" in what:
  if what.get("mapping"):
    if len(mapping_values) != 1:
      ...
  else:
    ...

if what.get("fallbackMapping") is not None and len(fallback_mapping_values) != 1:
  ...
```